crates/next-core/src/next_edge/entry.rs

```diff
@@ -18,10 +18,26 @@ pub async fn wrap_edge_entry(
     // The wrapped module could be an async module, we handle that with the proxy
     // here. The comma expression makes sure we don't call the function with the
     // module as the "this" arg.
+    // Turn exports into functions that are also a thenable. This way you can await the whole object
+    // or  exports (e.g. for Components) or call them directly as though they are async functions
+    // (e.g. edge functions/middleware, this is what the Edge Runtime does).
+    // Catch promise to prevent UnhandledPromiseRejectionWarning, this will be propagated through
+    // the awaited export(s) anyway.
     let source = formatdoc!(
         r#"
-            self._ENTRIES ||= {{}}
-            self._ENTRIES[{}] = import('MODULE')
+            self._ENTRIES ||= {{}};
+            const modProm = import('MODULE');
+            modProm.catch(() => {{}});
+            self._ENTRIES[{}] = new Proxy(modProm, {{
+                get(modProm, name) {{
+                    if (name === "then") {{
+                        return (res, rej) => modProm.then(res, rej);
+                    }}
+                    let result = (...args) => modProm.then((mod) => (0, mod[name])(...args));
+                    result.then = (res, rej) => modProm.then((mod) => mod[name]).then(res, rej);
+                    return result;
+                }},
+            }});
```

Comment:
That really looks like a workaround for something that can also be fixed by putting the `await` in the correct place.

Can't this be fixed in next.js code?

Reply:
This is a "workaround" for Vercel Edge calling `await self._ENTRIES["middleware_..."].default.call(...)`.

So this would have to be fixed somewhere in there, as opposed to Next.js.

Reply:
We need to fix that in the edge runtime